connector/connect/common/src/main/protobuf/spark/connect/relations.proto

```diff
@@ -598,3 +599,18 @@ message ToSchema {
   // The Sever side will update the dataframe with this schema.
   DataType schema = 2;
 }
+
+// Collect arbitrary (named) metrics from a dataset.
+message CollectMetrics {
+  // (Required) The input relation.
+  Relation input = 1;
+
+  // (Required) Name of the metrics.
+  string name = 2;
+
+  // (Required) The metric sequence.
+  repeated Expression metrics = 3;
+
+  // (Optional) The identity whether Observation are used.
+  optional bool is_observation = 4;
```

Comment:
So I have checked the code for how DF.observe works. In Scala it has two different overloads, one for Observation and one for string. Both end up calling the same underlying method on the dataframe. Both end up using the `CollectMetrics` and wrap it around the logical plan.

There is no need to have this special type for using the Observation. The simplification for Observation should be created on the client side.

Reply:
When using Observation, DF.observe cache the plan. The behavior is different. The flag used to tell server side whether cache the plan.

Reply:
OK

Reply:
Yeah you should be able to remove this.

Reply:
OK